File: app/src/main/java/com/simcom/ecashier/ui/addCollection/AddCollectionViewModel.java
package com.simcom.ecashier.ui.addCollection;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.simcom.ecashier.model.repositories.Repository;
import com.simcom.ecashier.model.room.Collection;


public class AddCollectionViewModel extends AndroidViewModel {

    private String name;
    //Only for testing!
    private int groupId = 1351;
    private int price;

    private Repository repository;

    public AddCollectionViewModel(Application application) {
        super(application);
        repository=new Repository(application);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getGroupId() {
        return groupId;
    }

    public void setGroupId(int groupId) {
        this.groupId = groupId;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public LiveData<Collection> getCurrentCollection(){
        return repository.getCurrentCollection();
    }
    public void finish(){
        Collection collection = new Collection(name,groupId,price,System.currentTimeMillis(), true);
        repository.addCollection(collection);
    }
}

File: app/src/main/java/com/simcom/ecashier/ui/addCollection/AddCollectionViewModel.kt
package com.simcom.ecashier.ui.addCollection

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import androidx.room.Room
import com.simcom.ecashier.model.room.CashierDatabase
import com.simcom.ecashier.model.room.Collection
import com.simcom.ecashier.model.room.Group
import com.simcom.ecashier.model.room.PersonToCollection
import kotlinx.coroutines.launch

class AddCollectionViewModel(application: Application) : AndroidViewModel(application) {
    var name: String? = null
    var groupId: Long? = null
    var price: Int? = null

    private var db:CashierDatabase = Room.databaseBuilder(
        application,
        CashierDatabase::class.java, "cashier_database"
    )
        .fallbackToDestructiveMigration()
        .build()

    fun createCollection() = viewModelScope.launch{
        if(name != null && groupId != null && price != null){
            val collection = Collection(name!!, groupId!!, price!!, System.currentTimeMillis(), true)
            db.collectionDao().makeNoCollectionCurrent()
            val collectionId = db.collectionDao().insert(collection)
            for(person in db.personToGroupDao().getPeopleFromGroupSynchronous(groupId!!)){
                db.personToCollectionDao().insert(PersonToCollection(collectionId, person.id, false, null))
            }
        }
    }

    fun getGroups() : LiveData<List<Group>> {
        return db.groupDao().allGroups()
    }


}